contracts/Template.sol

```diff
@@ -1,106 +1,129 @@
-        } else {
-            fac = _fac;
-        }
-    }
+    address constant ANY_ENTITY = address(-1);
 
-    function latestVersionAppBase(bytes32 appId) public view returns (address base) {
-        Repo repo = Repo(PublicResolver(ens.resolver(appId)).addr(appId));
-        (,base,) = repo.getLatest();
+    bool constant private TOKEN_TRANSFERABLE = true;
+    uint8 constant private TOKEN_DECIMALS = uint8(18);
+    uint256 constant private TOKEN_MAX_PER_ACCOUNT = uint256(0);            // no limit of tokens per account
 
-        return base;
-    }
-}
+    uint64 constant private DEFAULT_FINANCE_PERIOD = uint64(30 days);
 
+    mapping (address => address) internal tokenCache;
 
-contract Template is TemplateBase {
-    MiniMeTokenFactory tokenFactory;
+    constructor(DAOFactory _daoFactory, ENS _ens, MiniMeTokenFactory _miniMeFactory, IFIFSResolvingRegistrar _aragonID)
+        BaseTemplate(_daoFactory, _ens, _miniMeFactory, _aragonID)
+        public
+    {
+        _ensureAragonIdIsValid(_aragonID);
+        _ensureMiniMeFactoryIsValid(_miniMeFactory);
+    }
 
-    uint64 constant PCT = 10 ** 16;
-    address constant ANY_ENTITY = address(-1);
+    function newTokenAndInstance(
+        string _tokenName,
+        string _tokenSymbol,
+        string _id,
+        address[] _holders,
+        uint256[] _stakes,
+        uint64[3] _votingSettings, /* supportRequired, minAcceptanceQuorum, voteDuration */
+        uint64 _financePeriod,
+        bool _useAgentAsVault
+    )
+        external
+    {
+        newToken(_tokenName, _tokenSymbol);
+        newInstance(_id, _holders, _stakes, _votingSettings, _financePeriod, _useAgentAsVault);
+    }
 
-    constructor(ENS ens) TemplateBase(DAOFactory(0), ens) public {
-        tokenFactory = new MiniMeTokenFactory();
+    function newToken(string _name, string _symbol) public returns (MiniMeToken) {
+        MiniMeToken token = _createToken(_name, _symbol, TOKEN_DECIMALS);
+        _cacheToken(token, msg.sender);
+        return token;
     }
 
-    function newInstance() public {
-        Kernel dao = fac.newDAO(this);
-        ACL acl = ACL(dao.acl());
-        acl.createPermission(this, dao, dao.APP_MANAGER_ROLE(), this);
+    function newInstance(string _id, address[] _holders, uint256[] _stakes, uint64[3] _votingSettings, uint64 _financePeriod, bool _useAgentAsVault) public {
+        require(_holders.length > 0, ERROR_EMPTY_HOLDERS);
+        require(_holders.length == _stakes.length, ERROR_BAD_HOLDERS_STAKES_LEN);
+        require(_votingSettings.length == 3, ERROR_BAD_VOTE_SETTINGS);
+        MiniMeToken token = _popTokenCache(msg.sender);
+
+        // Create DAO and install apps
+        (Kernel dao, ACL acl) = _createDAO();
+        Vault agentOrVault = _useAgentAsVault ? _installDefaultAgentApp(dao) : _installVaultApp(dao);
+        Finance finance = _installFinanceApp(dao, agentOrVault, _financePeriod == 0 ? DEFAULT_FINANCE_PERIOD : _financePeriod);
```

Comment:
I wouldn’t bother with installing the Vault and Finance apps; generally they’re not useful for people (especially locally where there are no tokens deployed).